### PR TITLE
#71: Add patch to fix tidy extension and PHP 5.6.

### LINKED ISSUE #71: php56.extensions.tidy broken
```ShellSession
$ nix build .#php56.extensions.tidy -L
[…]
php-tidy> building
php-tidy> build flags: -j8 -l8 SHELL=/nix/store/pbfraw351mksnkp2ni9c4rkc9cpp89iv-bash-5.1-p12/bin/bash
php-tidy> /nix/store/pbfraw351mksnkp2ni9c4rkc9cpp89iv-bash-5.1-p12/bin/bash /build/php-5.6.40/ext/tidy/libtool --mode=compile gcc  -I. -I/build/php-5.6.40/ext/tidy -DPHP_ATOM_INC -I/build/php-5.6.40/ext/tidy/include -I/build/php-5.6.40/ext/tidy/main -I/build/php-5.6.40/ext/tidy -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/main -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/TSRM -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/Zend -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/ext -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/ext/date/lib -I/nix/store/67z2xvjmwial7fmii6cr8qpf4l334zyl-html-tidy-5.8.0/include  -DHAVE_CONFIG_H  -g -O2   -c /build/php-5.6.40/ext/tidy/tidy.c -o tidy.lo
php-tidy> mkdir .libs
php-tidy>  gcc -I. -I/build/php-5.6.40/ext/tidy -DPHP_ATOM_INC -I/build/php-5.6.40/ext/tidy/include -I/build/php-5.6.40/ext/tidy/main -I/build/php-5.6.40/ext/tidy -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/main -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/TSRM -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/Zend -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/ext -I/nix/store/agfgrrsh942wwmy2zbi2jf2rr63xj1pq-php-5.6.40-dev/include/php/ext/date/lib -I/nix/store/67z2xvjmwial7fmii6cr8qpf4l334zyl-html-tidy-5.8.0/include -DHAVE_CONFIG_H -g -O2 -c /build/php-5.6.40/ext/tidy/tidy.c  -fPIC -DPIC -o .libs/tidy.o
php-tidy> /build/php-5.6.40/ext/tidy/tidy.c:34:10: fatal error: buffio.h: No such file or directory
php-tidy>    34 | #include "buffio.h"
php-tidy>       |          ^~~~~~~~~~
php-tidy> compilation terminated.
php-tidy> make: *** [Makefile:181: tidy.lo] Error 1
```

Noticed in https://github.com/fossar/selfoss/runs/4924028211?check_suite_focus=true

### PR DESCRIPTION
**Pull request by drupol**
This PR:

- [x] Fix #71 
- [x] Add a new local patch (couldn't find a better online source)

### CODE REVIEW

File: pkgs/package-overrides.nix [hunk 1]
```diff
@@ -287,6 +287,20 @@ in
...
         throw "php.extensions.redis3 requires PHP version < 8.0.";
 
+    tidy = prev.extensions.iconv.overrideAttrs (attrs: {
+      patches =
+        let
+          upstreamPatches =
+            attrs.patches or [];
+
+          ourPatches =
+            lib.optionals (prev.php.version == "5.6") [
```
Comment (jtojnar) [inline pkgs/package-overrides.nix:297]: Would not the version have three components?

**Comment by jtojnar** [comment]
Would be nice to add it to the CI build list.

File: pkgs/package-overrides.nix (hunk 1 above)
Comment (drupol) [reply pkgs/package-overrides.nix:297]: I haven't build it for >= 5.6.

Comment (drupol) [reply pkgs/package-overrides.nix:297]: The patch is only needed for PHP 5.6. Check the tests.

File: pkgs/package-overrides.nix [hunk 2]
```diff
@@ -287,6 +287,20 @@ in
       else
         throw "php.extensions.redis3 requires PHP version < 8.0.";
 
+    tidy = prev.extensions.iconv.overrideAttrs (attrs: {
```
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/package-overrides.nix:290]: OMG

File: pkgs/package-overrides.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/package-overrides.nix:297]: Yeah, but the version would be 5.6.40, which would return false with `==`.

File: pkgs/package-overrides.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/package-overrides.nix]: Fixed.

**Review by jtojnar — APPROVED** [review]
LGTM, thanks.

**Pull request merged**